net-dns/knot/knot-2.5.3-r1.ebuild

```diff
@@ -0,0 +1,92 @@
+# Copyright 1999-2017 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+
+inherit eutils systemd user
+
+DESCRIPTION="High-performance authoritative-only DNS server"
+HOMEPAGE="http://www.knot-dns.cz/"
+SRC_URI="https://secure.nic.cz/files/knot-dns/${P/_/-}.tar.xz"
+
+LICENSE="GPL-3"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+
+KNOT_MODULES="+dnsproxy dnstap +noudp +onlinesign rosedb +rrl +stats +synthrecord +whoami"
+IUSE="doc caps +fastparser idn systemd +utils ${KNOT_MODULES}"
+
+RDEPEND="
+	>=net-libs/gnutls-3.3:=
+	>=dev-db/lmdb-0.9.15
+	dev-python/lmdb
+	>=dev-libs/userspace-rcu-0.5.4
+	caps? ( >=sys-libs/libcap-ng-0.6.4 )
+	dnstap? (
+		dev-libs/fstrm
+		dev-libs/protobuf-c
+	)
+	idn? ( || ( net-dns/libidn >=net-dns/libidn2-2.0.0 ) )
```

Comment:
This is almost certainly wrong. If you bind to the specific version of the library (i.e. link to it), you can't use `||` as that permits transparent replacement of one with the other. You either need to add a USE flag to switch between the two versions (and force the build system to use it), or just require the one that is preferred by the build system and forget about the other.